lib/transport-common.js

```diff
@@ -68,25 +68,23 @@ const newConnectAndInspectFn = (connFactory, transportClass) => {
     const callback = common.extractCallback(options);
     connect(appName, client, ...options, (error, connection) => {
       if (error) return callback(error);
-
-      Promise.all(interfaces.map(name => new Promise((resolve, reject) => {
-        connection.inspectInterface(name, (error, proxy) => {
-          if (error) {
-            reject(error);
-          } else {
-            resolve(proxy);
+      const proxies = {};
+      let errored = false;
+      const len = interfaces.length;
+      let cnt = 0;
+      interfaces.map(name => connection.inspectInterface(name,
```

Comment:
What's the point of using a `map` method here, if the result is being ignored in the end?
Maybe replace it with `forEach`?